### PR TITLE
to log out and then return

### PR DESCRIPTION
**Pull request by goofballLogic**
For #2

### CODE REVIEW

File: src/MvcClient.Portal/Startup.cs [hunk 1]
```diff
@@ -25,6 +25,8 @@ public void ConfigureServices(IServiceCollection services)
                     options.ResponseType = "code";
                     options.UsePkce = true;
                     options.ResponseMode = "query";
+                    options.SignedOutRedirectUri = "http://localhost:5002";
+                    options.SignedOutCallbackPath = "/";
```
Comment (haidery786) [inline src/MvcClient.Portal/Startup.cs]: Probably don't need to set  options.SignedOutCallbackPath, and options.SignedOutRedirectUri should be set to https and ideally the redirectUri should be configurable through appSettings.json

File: src/MvcClient.Portal/Controllers/HomeController.cs [hunk 2]
```diff
@@ -25,7 +25,8 @@ public IActionResult Index()
 
         public IActionResult Logout()
         {
-            return SignOut("cookie", "oidc");
+            Microsoft.AspNetCore.Authentication.AuthenticationProperties properties = new Microsoft.AspNetCore.Authentication.AuthenticationProperties { RedirectUri = "https://localhost:5002" };
+            return SignOut(properties, "cookie", "oidc");
         }
```
Comment (haidery786) [inline src/MvcClient.Portal/Controllers/HomeController.cs:33]: Yeah again RedirectUri value should be configurable.

File: src/IdenityServer4InMem/Views/Account/LoggedOut.cshtml [hunk 3]
```diff
@@ -3,6 +3,8 @@
 @{ 
     // set this so the layout rendering sees an anonymous user
     ViewData["signed-out"] = true;
+    Model.AutomaticRedirectAfterSignOut = true;
+    Model.PostLogoutRedirectUri = "https://localhost:5002";
```
Comment (haidery786) [inline src/IdenityServer4InMem/Views/Account/LoggedOut.cshtml]: this property "AutomaticRedirectAfterSignOut" may be set  to true when we build up LoggedOutViewModel object and may be there is way to set PostLogoutRedirectUri to redirectUri as part of configuration.

File: src/IdenityServer4InMem/Startup.cs [hunk 4]
```diff
@@ -54,7 +55,7 @@ public void ConfigureServices(IServiceCollection services)
                     // set the redirect URI to https://localhost:5001/signin-google
                     options.ClientId = "copy client ID from Google here";
                     options.ClientSecret = "copy client secret from Google here";
-                });
```
Comment (haidery786) [inline src/IdenityServer4InMem/Startup.cs]: looks like an empty line :)

File: src/IdenityServer4InMem/Startup.cs [hunk 5]
```diff
@@ -33,7 +33,8 @@ public void ConfigureServices(IServiceCollection services)
                 // see https://identityserver4.readthedocs.io/en/latest/topics/resources.html
                 options.EmitStaticAudienceClaim = true;
             })
-                .AddTestUsers(TestUsers.Users);
+                .AddTestUsers(TestUsers.Users)
+                ;
```
Comment (haidery786) [inline src/IdenityServer4InMem/Startup.cs]: semi colon on next line instead of end of statement

**Review by haidery786 — COMMENTED** [review]
Overall it is fine, just minor changes please. Thanks

File: src/MvcClient.Portal/Controllers/HomeController.cs [hunk 6]
```diff
@@ -9,10 +10,12 @@ namespace MvcClient.Portal.Controllers
...
     {
         private readonly ILogger<HomeController> _logger;
+        private readonly IConfiguration configuration;
 
-        public HomeController(ILogger<HomeController> logger)
+        public HomeController(ILogger<HomeController> logger, IConfiguration configuration)
         {
             _logger = logger;
+            this.configuration = configuration;
         }
```
Comment (haidery786) [inline src/MvcClient.Portal/Controllers/HomeController.cs:19]: do we need this keyword here, can we have configuration variable starting with underscore to keep consistency please.

**Review by haidery786 — APPROVED** [review]
Just one very minor thing to get rid of this keyword from constructor of HomeController. Thanks

File: src/IdenityServer4InMem/Views/Account/LoggedOut.cshtml (hunk 3 above)
Comment (goofballLogic) [reply src/IdenityServer4InMem/Views/Account/LoggedOut.cshtml]: I fixed everything I could, but can't get it working without this line.

File: src/MvcClient.Portal/Controllers/HomeController.cs (hunk 6 above)
Comment (goofballLogic) [reply src/MvcClient.Portal/Controllers/HomeController.cs:19]: thanks!

File: src/IdenityServer4InMem/Views/Account/LoggedOut.cshtml (hunk 3 above)
Comment (haidery786) [reply src/IdenityServer4InMem/Views/Account/LoggedOut.cshtml]: if you first of all set public static bool AutomaticRedirectAfterSignOut = true; in AcccountOptions.cs (IdSvr4 project) and add  // where to redirect to after logout
PostLogoutRedirectUris = { "https://localhost:5002/signout-callback-oidc" } on client in Config.cs (IdSvr4 project), then you don't need to have these lines here. Thanks

**Review by haidery786 — APPROVED** [review]

**Pull request merged**